infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/infrastructure/CorfuReplicationDiscoveryService.java

```diff
@@ -405,6 +404,12 @@ private void onLeadershipAcquire() {
         // check if all the sessions in system tables are valid
         sessionManager.removeStaleSessionOnLeadershipAcquire();
 
+
+        // If local cluster is Source, start the client config listener.
+        if (isSource()) {
+            sessionManager.startClientConfigListener();
```

Comment:
Yes good catch it could happen, although for LOGICAL_GROUP model there is unlikely to have such "role switchover" topology change, but we definitely need to handle it to be on the safe side. Added the related start/stop logic here.